### PR TITLE
Add image purchase plan to Azure driver on node_template resource

### PR DESCRIPTION
**Pull request by mouellet**
Adds plan attribute on the `node_template`'s `azure_config` for creating vms from marketplace images that have a cost.

Depends on: rancher/machine#122

Issues: 
- rancher/rancher#27854
- rancher/rancher#22927

### CODE REVIEW

**Comment by rawmind0** [comment]
Thanks for this PR @mouellet .  Waiting until PR https://github.com/rancher/machine/pull/122 is merged and released.

**Comment by snasovich** [comment]
This will need to be merged for 2.6.3 release as underlying changes are going to be in 2.6.3.

**Comment by brendarearden** [comment]
Available to test on v2.6.3-rc1

File: docs/resources/node_template.md [hunk 1]
```diff
@@ -178,6 +178,7 @@ The following attributes are exported:
 * `managed_disks` - (Optional) Configures VM and availability set for managed disks. Just for Rancher v2.3.x and above. Default `false` (bool)
 * `no_public_ip` - (Optional) Do not create a public IP address for the machine. Default `false` (bool)
 * `nsg` - (Optional) Azure Network Security Group to assign this node to (accepts either a name or resource ID, default is to create a new NSG for each machine). Default `docker-machine-nsg` (string)
+* `plan` - (Optional) Azure marketplace purchase plan for Azure Virtual Machine. Format is `<publisher>:<product>:<plan>`. (string)
```
Comment (rawmind0) [inline docs/resources/node_template.md]: Could you please add a comment here saying that `plan` argument is just for  Rancher v2.6.3 and above??

**Review by rawmind0 — CHANGES_REQUESTED** [review]

**Comment by brudnak** [comment]
# My checks PASSED

<hr>
  
## Validation Environment

| Component               | Version / Type                                     |
| ----------------------- | -------------------------------------------------- |
| Rancher version         | v2.6-fd930f76545c64fdad8b685d86090a8fbb8bebcf-head |
| Rancher cluster type    | Docker                                             |
| Docker version          | 20.10.7, build 20.10.7-0ubuntu5~20.04.2            |
| Helm version            | v2.16.8-rancher1                                   |
| Downstream cluster type | RKE1 Azure                                         |
| Downstream K8s version  | v1.21.7-rancher1-1                                 |
| Browser type            | Opera GX                                           |
| Browser version         | LVL3 (core: 82.0.4227.25) (x86_64)                 |

## Validation steps

<hr> 

#### Scenario 1 `PASS` | Create downstream Azure cluster with Purchase plan and image

1. Create an RKE1 Azure downstream cluster with:
    - Purchase Plan: `cloud-infrastructure-services:centos-8-3:centos-8-3`
    - Image: `cloud-infrastructure-services:centos-8-3:centos-8-3:0.0.1`


**RESULT**

Cluster successfully provisions. 

<hr>

#### Scenario 2 `PASS` | Create downstream Azure cluster without Purchase Plan and default image

1. Create an RKE1 Azure downstream cluster with:
    - Purchase Plan: leave this field blank
    - Default Image: `canonical:UbuntuServer:18.04-LTS:latest`


**RESULT**

Cluster successfully provisions. 

<hr>

### Additional Info

Other checks completed as well, all passing.

**Review by rawmind0 — APPROVED** [review]
Thanks for the update @mouellet 

LGTM

**Pull request merged**